commit ff1258e7ef39f1f1a69ef6475445712c1dc6e037
Author: Greg Farnum <gfarnum@redhat.com>
Date:   Fri Mar 10 18:48:47 2017 -0800

    mgr: introduce tick() calls through the system
    
    Still empty for the moment
    
    Signed-off-by: Greg Farnum <gfarnum@redhat.com>

diff --git a/src/mgr/Mgr.cc b/src/mgr/Mgr.cc
index b4892c3f8ef..158a8ac1a52 100644
--- a/src/mgr/Mgr.cc
+++ b/src/mgr/Mgr.cc
@@ -27,6 +27,7 @@
 #include "DaemonServer.h"
 #include "messages/MMgrBeacon.h"
 #include "messages/MMgrDigest.h"
+#include "messages/MMonMgrReport.h"
 #include "messages/MCommand.h"
 #include "messages/MCommandReply.h"
 #include "messages/MLog.h"
@@ -583,3 +584,14 @@ void Mgr::handle_mgr_digest(MMgrDigest* m)
   m->put();
 }
 
+void Mgr::tick()
+{
+  dout(0) << __func__ << dendl;
+  MMonMgrReport *m = new MMonMgrReport();
+  cluster_state.tick(m);
+  server.tick(m);
+  // TODO? We currently do not notify the PyModules
+  if (m->needs_send) {
+    monc->send_mon_message(m);
+  }
+}
diff --git a/src/mgr/Mgr.h b/src/mgr/Mgr.h
index 2efb2683f8a..0aa5b5c5be0 100644
--- a/src/mgr/Mgr.h
+++ b/src/mgr/Mgr.h
@@ -86,6 +86,8 @@ public:
 
   bool ms_dispatch(Message *m);
 
+  void tick();
+
   void background_init();
   void shutdown();
 };
